runtime/src/main/scala/org/apache/pekko/grpc/internal/Gzip.scala

```diff
@@ -24,13 +24,13 @@ object Gzip extends Codec {
   override def compress(uncompressed: ByteString): ByteString = {
     val baos = new ByteArrayOutputStream(uncompressed.size)
     val gzos = new GZIPOutputStream(baos)
-    try gzos.write(uncompressed.toArray)
+    try gzos.write(uncompressed.toArrayUnsafe())
     finally gzos.close()
     ByteString.fromArrayUnsafe(baos.toByteArray)
   }
 
   override def uncompress(compressed: ByteString): ByteString = {
-    val gzis = new GZIPInputStream(new ByteArrayInputStream(compressed.toArray))
+    val gzis = new GZIPInputStream(new ByteArrayInputStream(compressed.toArrayUnsafe()))
```

Comment:
Makes sense. We need to use MethodHandles to still support Pekko 1.0 (asInputStream is only in Pekko 1.1).

Something like https://github.com/apache/pekko-http/pull/539/files (this was reverted because of a bug in the hpack code - https://github.com/twitter/hpack/issues/43).

Reply:
> We need to use MethodHandles to still support Pekko 1.0 (asInputStream is only in Pekko 1.1).

thats unfortunate 

> this was reverted because of a bug in the hpack code - https://github.com/twitter/hpack/issues/43

last commit of that library was 5 years ago. also it [seems](https://github.com/twitter/hpack/issues/31) to be deprecated

Reply:
May be we can the netty 's Hpack？

Reply:
> We need to use MethodHandles to still support Pekko 1.0 (asInputStream is only in Pekko 1.1).

IMHO we could just require Pekko 1.1 in Pekko gRPC 1.1

Reply:
@raboof I got rid of the MethodHandle in https://github.com/apache/pekko-grpc/pull/309